Code/Tools/ProjectManager/Source/Settings.cpp

```diff
@@ -0,0 +1,201 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Settings.h>
+
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/IO/ByteContainerStream.h>
+#include <AzCore/Utils/Utils.h>
+
+namespace O3DE::ProjectManager
+{
+    Settings::Settings(bool saveToDisk)
+        : m_saveToDisk(saveToDisk)
+    {
+        m_settingsRegistry = AZ::SettingsRegistry::Get();
+
+        AZ_Assert(m_settingsRegistry, "Failed to create Settings");
+    }
+
+    bool Settings::IsInitialized()
+    {
+        // Settings intialized correctly if it successfuly got the SettingsRegistry
+        return m_settingsRegistry;
+    }
+
+    bool Settings::Save()
+    {
+        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+        dumperSettings.m_prettifyOutput = true;
+        dumperSettings.m_jsonPointerPrefix = ProjectManagerKeyPrefix;
+
+        AZStd::string stringBuffer;
+        AZ::IO::ByteContainerStream stringStream(&stringBuffer);
+        if (!AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(
+                *m_settingsRegistry, ProjectManagerKeyPrefix, stringStream, dumperSettings))
+        {
+            AZ_Warning("ProjectManager", false, "Could not save Project Manager settings to stream");
+            return false;
+        }
+
+        AZ::IO::FixedMaxPath o3deUserPath = AZ::Utils::GetO3deManifestDirectory();
+        o3deUserPath /= AZ::SettingsRegistryInterface::RegistryFolder;
+        o3deUserPath /= "ProjectManager.setreg";
+
+        bool saved = false;
+        constexpr auto configurationMode =
+            AZ::IO::SystemFile::SF_OPEN_CREATE | AZ::IO::SystemFile::SF_OPEN_CREATE_PATH | AZ::IO::SystemFile::SF_OPEN_WRITE_ONLY;
+
+        AZ::IO::SystemFile outputFile;
+        if (outputFile.Open(o3deUserPath.c_str(), configurationMode))
+        {
+            saved = outputFile.Write(stringBuffer.data(), stringBuffer.size()) == stringBuffer.size();
+        }
+
+        AZ_Warning("ProjectManager", saved, "Unable to save Project Manager registry file to path: %s", o3deUserPath.c_str());
+        return saved;
+    }
+
+    bool Settings::Get(QString& result, const QString& settingsKey)
+    {
+        bool success = false;
+
+        AZStd::string settingsValue;
+        success = m_settingsRegistry->Get(settingsValue, settingsKey.toStdString().c_str());
+
+        result = settingsValue.c_str();
+        return success;
+    }
+
+    bool Settings::Get(bool& result, const QString& settingsKey)
+    {
+        return m_settingsRegistry->Get(result, settingsKey.toStdString().c_str());
+    }
+
+    bool Settings::Set(const QString& settingsKey, const QString& settingsValue)
+    {
+        bool success = false;
+
+        success = m_settingsRegistry->Set(settingsKey.toStdString().c_str(), settingsValue.toStdString().c_str());
+        if (m_saveToDisk)
+        {
+            Save();
+        }
```

Comment:
minor: you could move this common if(blah) save() block into an `OnSettingsUpdated` private method

Reply:
Sounds like a good idea!